app/controllers/admin/adventure_missions_controller.rb

```diff
@@ -0,0 +1,80 @@
+class Admin::AdventureMissionsController < Admin::BaseController
+  before_action :set_introduction_and_adventure_from_url, only: [:index, :new, :create]
+  before_action :set_introduction_and_adventure_from_object, only: [:show, :edit, :update, :destroy]
+  load_permissions_and_authorize_resource through: :adventure
+
+  def index
+    @grid = initialize_grid(@adventure_missions, locale: :sv, order: 'adventure_missions.index')
+  end
+
+  def new
+    @adventure_mission = AdventureMission.new
+    @suggested_index = @adventure.adventure_missions.count
+  end
+
+  def create
+    @adventure_mission = AdventureMission.new(adventure_mission_params)
+
+    if @adventure.adventure_missions.pluck(:index).include?(adventure_mission_params[:index].to_i)
```

Comment:
Move this to a service

Reply:
Also, what does it do?

Reply:
Fixes so that if a taken index is chosen the other indexes are updated. Don't know if this is the best way to do it though

Reply:
I think you can do this in a very simple SQL query